scripts/src/public/services/orders.ts

```diff
@@ -276,12 +304,15 @@ export async function submitOrder(
 	await order.save();
 	logger.info("order changed to pending", { orderId });
 
-	if (order.type === "earn") {
```

Comment:
Just because of this check, I think that the order should contain the nature of the flow and not infer it.

Either add the origin/order_type into order or down here check that there is only a single context and that it is earn (instead of checking for existence of earn)